Sources/SpeziSecureStorage/SecureStorage.swift

```diff
@@ -60,111 +58,42 @@ public final class SecureStorage: Module, DefaultInitializable, EnvironmentAcces
     /// - Parameter tag: The tag used to identify the key in the keychain or the secure enclave.
     /// - Returns: Returns the private `SecKey` generated and stored in the keychain or the secure enclave.
     public func retrievePrivateKey(forTag tag: String) throws -> SecKey? {
-        // This method follows
-        // https://developer.apple.com/documentation/security/certificate_key_and_trust_services/keys/storing_keys_in_the_keychain
-        // for guidance.
-        
-        var item: CFTypeRef?
-        do {
-            try execute(SecItemCopyMatching(keyQuery(forTag: tag) as CFDictionary, &item))
-        } catch SecureStorageError.notFound {
-            return nil
-        } catch {
-            throw error
-        }
-        
-        // Unfortunately we have to do a force cast here.
-        // The compiler complains that "Conditional downcast to CoreFoundation type 'SecKey' will always succeed"
-        // if we use `item as? SecKey`.
-        return (item as! SecKey) // swiftlint:disable:this force_cast
+        try keyStorage.retrievePrivateKey(forTag: tag)
     }
     
     /// Retrieves a public key stored in the keychain or the secure enclave identified by a `tag`.
     /// - Parameter tag: The tag used to identify the key in the keychain or the secure enclave.
     /// - Returns: Returns the public `SecKey` generated and stored in the keychain or the secure enclave.
     public func retrievePublicKey(forTag tag: String) throws -> SecKey? {
-        guard let privateKey = try retrievePrivateKey(forTag: tag),
-              let publicKey = SecKeyCopyPublicKey(privateKey) else {
-            return nil
-        }
-        
-        return publicKey
+        try keyStorage.retrievePublicKey(forTag: tag)
     }
     
     /// Deletes the key stored in the keychain or the secure enclave identified by a `tag`.
     /// - Parameter tag: The tag used to identify the key in the keychain or the secure enclave.
     public func deleteKeys(forTag tag: String) throws {
-        do {
-            try execute(SecItemDelete(keyQuery(forTag: tag) as CFDictionary))
-        } catch SecureStorageError.notFound {
-            return
-        } catch {
-            throw error
-        }
-    }
-    
-    private func keyQuery(forTag tag: String) -> [String: Any] {
-        var query: [String: Any] = [
-            kSecClass as String: kSecClassKey,
-            kSecAttrApplicationTag as String: tag,
-            kSecAttrKeyType as String: kSecAttrKeyTypeECSECPrimeRandom,
-            kSecReturnRef as String: true
-        ]
-        
-        #if os(macOS)
-        query[kSecUseDataProtectionKeychain as String] = true
-        #endif
-        
-        return query
+        try keyStorage.delete(forTag: tag)
     }
     
     
     // MARK: - Credentials Handling
     
+    @available(*, deprecated, renamed: "store(credential:server:removeDuplicate:storageScope:)")
```

Comment:
Nice, thank you fror the deprivation warnings to allow us to tag a minor version!